units/provisioning_user_data_done.go

```diff
@@ -0,0 +1,144 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+)
+
+const (
+	userDataDoneJobName = "user-data-done"
+)
+
+type userDataDoneJob struct {
+	HostID   string `bson:"host_id" json:"host_id" yaml:"host_id"`
+	job.Base `bson:"base" json:"base" yaml:"base"`
+
+	host *host.Host
+	env  evergreen.Environment
+}
+
+func init() {
+	registry.AddJobType(userDataDoneJobName, func() amboy.Job {
+		return makeUserDataDoneJob()
+	})
+}
+
+func makeUserDataDoneJob() *userDataDoneJob {
+	j := &userDataDoneJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    userDataDoneJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+// NewUserDataDoneJob creates a job that checks if the host is done running its
+// user data if bootstrapped with user data.
+func NewUserDataDoneJob(env evergreen.Environment, h host.Host, id string) amboy.Job {
+	j := makeUserDataDoneJob()
+	j.host = &h
+	j.HostID = h.Id
+	j.env = env
+	j.SetPriority(1)
+	j.SetID(fmt.Sprintf("%s.%s.%s", userDataDoneJobName, j.HostID, id))
+
+	return j
+}
+
+func (j *userDataDoneJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	if err := j.populateIfUnset(); err != nil {
+		j.AddError(err)
+		return
+	}
+
+	if j.host.Status != evergreen.HostProvisioning {
+		grip.Info(message.Fields{
+			"message": "skipping user data check because host is no longer provisioning",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		})
```

Comment:
do we need to log this much? errors from jobs are logged already by amboy, the extra data is nice, Maybe, but seems exessive.